sunpy/net/hek/hek.py

```diff
@@ -17,30 +15,66 @@
+from sunpy.net.hek.utils import freeze, parse_times, parse_values_to_quantities
 from sunpy.util import dict_keys_same, unique
 from sunpy.util.xml import xml_to_dict
 
 __all__ = ['HEKClient', 'HEKTable', 'HEKRow']
 
 DEFAULT_URL = 'https://www.lmsal.com/hek/her?'
 
-
-def _freeze(obj):
-    """ Create hashable representation of result dict. """
-    if isinstance(obj, dict):
-        return tuple((k, _freeze(v)) for k, v in obj.items())
-    if isinstance(obj, list):
-        return tuple(_freeze(elem) for elem in obj)
-    return obj
-
-
 class HEKClient(BaseClient):
     """
     Provides access to the Heliophysics Event Knowledgebase (HEK).
 
     The HEK stores solar feature and event data generated by algorithms and
     human observers.
+
+    Notes
+    -----
+
+    It is important to be aware that the client transforms the raw outputs from the HEK.
+    This means that it will add units to values, or return astropy Time instead of string.
+
+    That entails the merging or removal of columns that have no purpose anymore.
+    For example, columns ``event_coord1``, ``event_coord2`` and ``event_coord3``
+    are merged into one new column called ``event_coord`` and returns a
+    `~astropy.coordinates.SkyCoord`.
+
+    Furthermore, we create following coordinate systems corresponding to each the following columns:
+
+    * `~astropy.coordinates.ICRS`.
+
+        * ``bound_chaincode``: Provides coordinates in the ICRS frame using boundary chaincode.
+        * ``skel_chaincode``: Provides coordinates in the ICRS frame using skeleton chaincode.
+
+    * `~sunpy.coordinates.frames.HeliographicCarrington`.
+
+        * ``hgc_bbox``: Represents the bounding box coordinates in the Heliographic Carrington (HGC) coordinate system.
+        * ``hgc_coord``: Provides specific coordinates in the HGC coordinate system.
+        * ``hgc_boundcc``: Uses boundary chaincode to provide coordinates in the HGC coordinate system.
+
+    * `~sunpy.coordinates.Helioprojective`.
+
+        * ``hpc_bbox``: Provides bounding box coordinates in the Helioprojective coordinate system (HPC).
+        * ``hpc_coord``: Specifies coordinates in the HPC frame.
+        * ``hpc_boundcc``: Uses boundary chaincode to specify coordinates in the HPC coordinate system.
+
+    * `~sunpy.coordinates.HeliographicStonyhurst`.
+
+        * ``hgs_bbox``: Represents the bounding box coordinates in the Heliographic Stonyhurst (HGS) coordinate system.
+        * ``hgs_coord``: Provides specific coordinates in the HGS coordinate system.
+        * ``hgs_boundcc``: Uses boundary chaincode to provide coordinates in the HGS coordinate system.
+
+    * `~sunpy.coordinates.Heliocentric`.
+
+        * ``hrc_bbox``: Represents the bounding box coordinates in the Heliocentric coordinate system (HRC).
+        * ``hrc_coord``: Specifies coordinates in the HRC frame.
+        * ``hrc_boundcc``: Uses boundary chaincode to specify coordinates in the HRC coordinate system.
```

Comment:
Does the abbreviation "HRC"/"hrc" come from HEK?  I ask because we use "HCC" in SunPy (i.e., for HelioCentric Cartesian).  I don't know why they would stick an "R" in there...